components/bt/host/bluedroid/stack/btm/btm_inq.c

```diff
@@ -2537,7 +2537,69 @@ void BTM_AddEirService( UINT32 *p_eir_uuid, UINT16 uuid16 )
 
 /*******************************************************************************
 **
-** Function         BTM_RemoveEirService
+** Function         btm_find_custom_uuid_slots
+**
+** Description      Helper function for custom service managing routines.
+**
+** Parameters       custom_uuid - pointer to custom_uuid array in tBTA_DM_CB
+**                  uuid - UUID struct
+**
+** Returns          None
+**
+*******************************************************************************/
+static void btm_find_custom_uuid_slots(tBT_UUID *custom_uuid, tBT_UUID uuid,
+                                         UINT8 *empty_slot, UINT8 *match_slot)
+{
+    *empty_slot = BTA_EIR_SERVER_NUM_CUSTOM_UUID;
+    *match_slot = BTA_EIR_SERVER_NUM_CUSTOM_UUID;
+
+    for ( UINT8 xx = 0; xx < BTA_EIR_SERVER_NUM_CUSTOM_UUID; xx++ ) {
+        if ( custom_uuid[xx].len == 0 ) {
+            if ( *empty_slot == BTA_EIR_SERVER_NUM_CUSTOM_UUID )
+                *empty_slot = xx;
+        } else if ( *match_slot == BTA_EIR_SERVER_NUM_CUSTOM_UUID ) {
+            if ( !memcmp (&custom_uuid[xx].uu, &uuid.uu, uuid.len) ) {
+                *match_slot = xx;
```

Comment:
If it has been matched, can it be returned directly?

Reply:
Yes.
I split this search algorithm into `btm_find_empty_custom_uuid_slot()` and `btm_find_match_custom_uuid_slot()`.
This should make the code simpler.